protocol_rfcs/collated-string-type.md

```diff
@@ -0,0 +1,146 @@
+
+Collations can be specified for all string fields in a table schema. It is also possible to store statistics per collation version. This is required because the min and max values of a column can differ based on the used collation or collation version.
+
+By default, all strings are collated using binary collation. That means that strings compare equal if their binary representations are equal. The binary representation is also used to sort them.
+
+#### Collation identifiers
+
+Collations can be referred to using collation identifiers. The Delta format does not specify any collation rules other than binary collation, but supports the concept of collation providers such that engines can use providers like [ICU](https://icu.unicode.org/) and mark statistics accordingly.
+
+A collation identifier consists of 3 parts, which are combined into one identifier using dots as separators. Dots are not allowed to be part of provider and collation names, but can be used in versions.
+
+Part | Description
+-|-
+Provider | Name of the provider. Must not contain dots
+Name | Name of the collation as provided by the provider. Must not contain dots
+Version | Version string. Is allowed to contain dots
+
+#### Specifying collations in the table schema
+
+Collations can be specified for all string types in a schema. This includes string fields, but also the key and value type of maps and the element type of arrays. Collations are specified in the metadata of the closest enclosing field.
+Collation identifiers are stored in a `collations` object. These object can have 4 keys:
+
+Key | Value | Description
+-|-|-
+collation | collation identifier | Collation of a string field. Only valid when none of the other keys are present.
+elementCollation | `collations object` | Collations of elements in an array. Only valid when none of the other keys are present.
+keyCollation | `collations object` | Collations for the key type of a map. Only valid on it's own or with `valueCollation`.
+valueCollation | `collations object` | Collations for the value type of a map. Only valid on it's own or with `keyCollation`.
+
+This example provides an overview of how collations are stored. Note that irrelevant fields have been stripped.
+
+```
+{
+  "type" : "struct",
+  "fields" : [ {
+    "name" : "col1",
+    "type" : "map",
+    "keyType": "string"
+    "valueType": {
+      "type": "array"
+      "elementType": {
+        "type": "map"
+        "keyType: {
+          "type": "array"
+          "elementType": "string"
+        },
+        "valueType": {
+          "type": "map",
+          "keyType": "string",
+          "valueType": {
+            "type": "struct",
+            "fields": [ {
+              "name": "f1"
+              "type": "string"      
+            } ],
+          },
+          metadata: {
+            "collations": { "collation": "ICU.de_DE.73" }
```

Comment:
why is version in the schema?

also I think we agreed to a slightly different metadata format: https://github.com/apache/spark/pull/46280